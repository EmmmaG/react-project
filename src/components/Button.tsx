import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import ButtonComponent from '@material-ui/core/Button'

const styles = (theme: Theme) => ({
	/* eslint-disable quote-props */
	button: {
		color: theme.palette.grey[500],
		backgroundColor: 'white',
		fontSize: '13px',
		textTransform: 'none' as 'none',
		fontWeight: 700,
		padding: '3px 20px',
		borderRadius: '3px',
		marginRight: '10px',
		marginBottom: '10px',
		border: `1px solid ${theme.palette.grey[500]}`,
		'&:hover': {
			background: theme.palette.grey[600],
			color: 'white',
		},
	},
	/* eslint-enable quote-props */
})

export interface Props extends WithStyles<typeof styles> {
    buttonContent: string
    link: string
}

export class Button extends React.Component<Props> {
	private onClick = () => {
		window.open(
			this.props.link,
			'_blank',
		)
	}
	public render() {
		const { classes, buttonContent } = this.props

		return (
			<ButtonComponent
				className={classes.button}
				onClick={this.onClick}
			>
				{buttonContent}
			</ButtonComponent>
		)
	}
}

export default withStyles(styles)(Button)
