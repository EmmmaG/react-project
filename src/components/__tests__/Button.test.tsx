import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import Button, { Props } from '../Button'

const props: Props = {
	buttonContent: 'button test',
	link: 'www.google.com',
	classes: idObj,
}

describe('Button', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Button {...props} />)
		expect(wrapper).toMatchSnapshot()
	})
})
