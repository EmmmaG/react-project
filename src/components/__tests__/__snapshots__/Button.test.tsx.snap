// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Button
  buttonContent="button test"
  classes={
    Object {
      "button": "Button-button-1",
    }
  }
  link="www.google.com"
/>
`;
